cluster_management/src/main/java/com/pinterest/rocksplicator/ConfigGenerator.java

```diff
@@ -267,19 +304,24 @@ private String getHostWithDomain(String host) {
 
   // update disabledHosts, return true if there is any changes
   private boolean updateDisabledHosts() {
-    HelixAdmin admin = helixManager.getClusterManagmentTool();
+    this.updateLock.lock();
```

Comment:
Which fields does this lock prorect? I was under the impression that the event callbacks won't be called concurrently by helix framework, thus no application side synchronization is required.

Reply:
I am not so sure about what helix framework will do when there are different event type notifications. In any case adding application side lock should be a no-op if helix framework does guarantee delivery of all event notifications in singe thread. If it doesn't, then above help prevent race condition.

Reply:
I remembered I verified that Helix use single thread to run these callbacks. I'd prefer to not add it for code simplicity.

But if you still want to add this lock, can you add it to the top level of the callback functions? and make the lock name explicit that this lock is to synchronize callback functions. Otherwise, people may need to guess which fields this lock will protect.

Reply:
Done.

Reply:
can we add these discussion and explanation as part of the comment to help people understand it. ty

Reply:
Added.